### DISCUSSION TITLE
How to substitute storage with Azure CosmosDB?

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#13865] [Q&A] [answered]
I have the other stuff stored in ComsosDB, so I want CMS data to live there too.
How to achieve that?
What interfaces do I need to implement for that?

**Comment by hishamco** [comment]
I think you need to add your own DB Provider that extends YesSQL data provider, have a look to SqlServer implementation https://github.com/sebastienros/yessql/tree/main/src/YesSql.Provider.SqlServer

**Reply by xperiandri** [reply]
But can I substitute YesSql altogether?
@sebastienros

**Reply by hishamco** [reply]
No need to replace it, you need to inject your custom provider like the default one, but I'm not sure if the `AddDataAccess()` is extensible to add a new provider instead of rewriting the logic

Waiting for Seb ...

**Reply by xperiandri** [reply]
@sebastienros any help with this?

**Reply by ns8482e** [reply]
I guess probably- need to replace ootb contentmanager with new one that do not use yessql

**Reply by xperiandri** [reply]
Could you point to the code file?

**Reply by sebastienros** [reply]
Multiple components use YesSql directly and build SQL queries. I don't think this will be possible. You can have some components use something else but you won't be able to replace yessql completely. Not until any data access is actually. One good example of such abstraction is the user management which uses UserStores (from ASP.NET Identity) and could have custom implementations.

**Comment by erinloy** [comment, marked as answer]
https://github.com/erinloy/YesSql.Provider.CosmosDb
https://www.nuget.org/packages/YesSql.Provider.CosmosDb

**Reply by Piedone** [reply]
Oh, very nice!

**Comment by hishamco** [comment]
I worked recently on CosmosDB, but it's great to see someone created a YesSQL provider for that

**Comment by DenisBalan** [comment]
fyi https://github.com/erinloy/YesSql.Provider.CosmosDb/issues/1
